hypothesis-python/src/hypothesis/_strategies.py

```diff
@@ -239,7 +239,7 @@ def calc_is_empty(self, recur):
     def do_draw(self, data):
         # This method should never be called because draw() will mark the
         # data as invalid immediately because is_empty is True.
-        assert False  # pragma: no cover
+        raise NotImplementedError  # pragma: no cover
```

Comment:
Nit: I'm not wild about `raise ExceptionType` syntax and prefer `raise ExceptionType()`